userspace/otpilot/src/main.rs

```diff
@@ -82,21 +84,13 @@ fn run() -> TockResult<()> {
         device_id: [0; 64],
     };
 
-    {
-        let mut idx : usize = 0;
-        for val in "v1.00".as_bytes() {
-            if idx > identity.version.len() { break; }
-            identity.version[idx] = *val;
-            idx = idx + 1;
-        }
-    }
+    let banner_bytes = "v1.00".as_bytes();
+    let max_len = min(identity.version.len(), banner_bytes.len());
```

Comment:
If this `min()` call is really needed, then we should print an error message.  I'd be happy with an assertion here as well.  But silent truncation is a bad idea, IMESHO.

Reply:
Fair. I added code to print an error.